bundles/model/org.eclipse.smarthome.model.sitemap/src/org/eclipse/smarthome/model/sitemap/SitemapProvider.java

```diff
@@ -14,21 +14,37 @@
 
 import java.util.Set;
 
+import org.eclipse.smarthome.model.core.ModelRepositoryChangeListener;
 
 public interface SitemapProvider {
 
     /**
      * This method provides access to sitemap model files, loads them and returns the object model tree.
-     * 
+     *
      * @param sitemapName the name of the sitemap to load
      * @return the object model tree, null if it is not found
      */
     public Sitemap getSitemap(String sitemapName);
 
     /**
      * Returns the names of all available sitemaps
-     * 
+     *
      * @return names of provided sitemaps
      */
     public Set<String> getSitemapNames();
+
+    /**
+     * Add a listener which will be informed subsequently once a model has changed
+     *
+     * @param listener
+     */
+    public void addModelChangeListener(ModelRepositoryChangeListener listener);

```

Comment:
I know the other methods in this interface also have the `public` keyword, but they are superfluous in an interface...